pkg/externalresource/external_resource.go

```diff
@@ -0,0 +1,58 @@
+package externalresource
+
+import (
+	b64 "encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+// ExternalResources represents the map of external resources at a manifest
+type ExternalResources map[string]*ExternalResource
```

Comment:
Let's use a name that can differentiate more from the actual type

```suggestion
type ExternalResourceSection map[string]*ExternalResource
```